src/pkg/boltz.ts

```diff
@@ -0,0 +1,465 @@
+  type: 'submarine' | 'reversesubmarine';
+  pairId: 'L-BTC/BTC';
+  orderSide: 'buy' | 'sell';
+}
+
+interface CreateSwapCommonResponse {
+  id: string;
+  timeoutBlockHeight: number;
+}
+
+interface SubmarineSwapRequest {
+  invoice: string;
+  refundPublicKey: string;
+}
+
+interface SubmarineSwapResponse {
+  acceptZeroConf: boolean;
+  address: string;
+  bip21: string;
+  blindingKey: string; // private
+  expectedAmount: number;
+  redeemScript: string;
+}
+
+interface ReverseSubmarineSwapRequest {
+  preimageHash: string;
+  invoiceAmount: number;
+  claimPublicKey: string;
+}
+
+interface ReverseSubmarineSwapResponse {
+  blindingKey: string; // private
+  invoice: string;
+  lockupAddress: string;
+  onchainAmount: number;
+  redeemScript: string;
+}
+
+export interface SubmarineSwap {
+  address: string;
+  expectedAmount: number;
+  id: string;
+  redeemScript: string;
+}
+
+export interface ReverseSwap {
+  blindingPrivateKey: string;
+  claimPublicKey: string;
+  id: string;
+  invoice: string;
+  lockupAddress: string;
+  preimage: Buffer;
+  redeemScript: string;
+  timeoutBlockHeight: number;
+}
+
+interface MakeClaimTransactionParams {
+  utxo: Unspent;
+  claimKeyPair: ECPairInterface;
+  preimage: Buffer;
+  redeemScript: Buffer;
+  destinationScript: Buffer;
+  fee: number;
+  blindingPublicKey: Buffer;
+  timeoutBlockHeight: number;
+}
+
+const feeAmount = 500; // fee for regular liquid tx
+const swapFeeAmount = 500; // fee for Boltz
+// lightning swap invoice amount limit (in satoshis)
+export const DEFAULT_LIGHTNING_LIMITS = { maximal: 4294967, minimal: 50000 };
+export const DEPOSIT_LIGHTNING_LIMITS = {
+  maximal: DEFAULT_LIGHTNING_LIMITS.maximal - feeAmount - swapFeeAmount,
+  minimal: DEFAULT_LIGHTNING_LIMITS.minimal - feeAmount - swapFeeAmount,
+};
+
+export const boltzUrl: Record<NetworkString, string> = {
+  regtest: 'http://localhost:9090',
+  testnet: 'https://testnet.boltz.exchange/api',
+  liquid: 'https://boltz.exchange/api',
+};
+
+export class Boltz implements BoltzInterface {
+  private asset: string;
+  private url: string;
+  private zkp: ZKP;
+
+  constructor(url: string, asset: string, zkp: ZKP) {
+    this.asset = asset;
+    this.url = url;
+    this.zkp = zkp;
+  }
+
+  async getBoltzPair(pair: string): Promise<any> {
+    const data = await this.getApi(`${this.url}/getpairs`);
+    if (!data?.pairs?.[pair]) return;
+    return data.pairs[pair];
+  }
+
+  // return invoice expire date
+  getInvoiceExpireDate(invoice: string): number {
+    const { timeExpireDate } = bolt11.decode(invoice);
+    return timeExpireDate ? timeExpireDate * 1000 : 0; // milliseconds
+  }
+
+  // return value in given invoice
+  getInvoiceValue(invoice: string): number {
+    const { satoshis, millisatoshis } = bolt11.decode(invoice);
+    if (satoshis) return fromSatoshi(satoshis, 8);
+    if (millisatoshis) return fromSatoshi(Number(millisatoshis) / 1000, 8);
+    return 0;
+  }
+
+  makeClaimTransaction({
```

Comment:
this function does not call any external services, it should let to write a regtest unit test using a "fake" utxo? this test could check if the pset created is correctly done (is blinded, is signed with preimage etc...)